lib/jobs/upsert-node-job.js

```diff
@@ -0,0 +1,41 @@
+// Copyright 2016, EMC, Inc.
+/* jshint node:true */
+
+'use strict';
+
+module.exports = UpsertNodeAttributeFactory;
+
+var di = require('di');
+
+di.annotate(UpsertNodeAttributeFactory, new di.Provide('Job.Upsert.Node.Attribute'));
+di.annotate(UpsertNodeAttributeFactory, new di.Inject(
+    'Job.Base',
+    'Util',
+    'Services.Waterline',
+    'Logger'
+    )
+);
+
+function UpsertNodeAttributeFactory(BaseJob, util, waterline, Logger) {
+    var logger = Logger.initialize(UpsertNodeAttributeFactory);
+
+    function UpsertNodeAttribute(options, context, taskId) {
+        UpsertNodeAttribute.super_.call(this, logger, options, context, taskId);
+        this.nodeId = this.options.target;
+        this.data = this.options.data;
+    }
+    util.inherits(UpsertNodeAttribute, BaseJob);
+    
+    UpsertNodeAttribute.prototype._run = function run() {
+        var self = this;
+        return waterline.nodes.updateByIdentifier(self.nodeId, self.data) 
```

Comment:
@zhangy42 I think @yyscamper is correct.  updateByIdentifier should throw a NotFoundError if the node does not exist.  There are a few ways to do an upsert.  One is to use a native mongo update with the `upsert` option.  Another options is to create an upsert by chaining a waterline update and create together.True upserts also require an index on the primary key to prevent duplicate records from getting inserted.